pkg/discovery/oci_dbbacked.go

```diff
@@ -133,8 +138,32 @@ func (od *DBBackedOCIDiscovery) listGroupsFromInventory() ([]*plugininventory.Pl
 // inventory of this discovery and stores it in the cache directory.
 func (od *DBBackedOCIDiscovery) fetchInventoryImage() error {
 	// TODO(khouzam): Improve by checking if we really need to download again or if we can use the cache
+
+	// Get the custom public key path and prepare cosign verifier, if empty, cosign verifier would use embedded public key for verification
+	customPublicKeyPath := os.Getenv(constants.CustomPublicKeyPathForPluginMetadataSignature)
+	cosignVerifier := cosignhelper.NewCosignVerifier(customPublicKeyPath)
+	if err := od.verifyInventoryImageSignature(cosignVerifier); err != nil {
+		log.Warningf("Unable to verify Plugin metadata signature: %v", err)
+		log.Warningf("Warning, Plugin metadata signature verification failed and therefore CLI couldn't ensure the integrity of the plugins to be installed. If you choose to ignore the validation please set environment variable SKIP_PLUGIN_METADATA_SIGNATURE_VERIFY to true ")
```

Comment:
>  think this is an important message. I made the below suggestion but I would like us to agree on the proper warning. 
Thanks @marckhouzam for the suggestion. Yes, we should discuss and agree on the proper warning.  I would be glad to hear suggestions.

Reply:
Update: CLI would fail always and would show the below message when the discovery image signature verification fails. 

Sample log:
```
❯ ./bin/tanzu plugin search
[!] Unable to verify the plugins discovery image signature: failed unmarshalling PEM encoded default public key: PEM decoding failed
[x] Fatal, plugins discovery image signature verification failed. The `tanzu` CLI can not ensure the integrity of the plugins to be installed. To ignore this validation please set the environment variable "TANZU_CLI_SKIP_PLUGIN_DISCOVERY_IMAGE_SIGNATURE_VERIFICATION" to true.  This is NOT RECOMMENDED and could put your environment at risk!
```

Reply:
Update: After discussion, the proposed global skip verification environment variable `TANZU_CLI_SKIP_PLUGIN_DISCOVERY_IMAGE_SIGNATURE_VERIFICATION` is changed to `TANZU_CLI_PLUGIN_DISCOVERY_IMAGE_SIGNATURE_VERIFICATION_SKIP_LIST` and would contain a comma separated list of plugin discovery image urls

ex: `TANZU_CLI_PLUGIN_DISCOVERY_IMAGE_SIGNATURE_VERIFICATION_SKIP_LIST=harbor-repo.vmware.com/tanzu_cli/pkalle/test/v1/plugin-inventory:latest,harbor-repo.vmware.com/tanzu_cli/pkalle/plugins/plugin-inventory:latest`